### ISSUE TITLE
ninja: error: loading 'build.ninja': No such file or directory

### ISSUE THREAD

**Issue by MRYangY** [ninja-build/ninja#1463]
environment :Mac  
Install ninja with the brew intsall ninja command
then on terminal execute ninja occur ninja: error: loading 'build.ninja': No such file or directory 。
➜  ~ ninja 
ninja: error: loading 'build.ninja': No such file or directory
➜  ~ 

Why is this problem still occurring when you have successfully installed ninja?

**Comment by MRYangY** [comment]
@mydongistiny yes !  Need to take parameters?

**Comment by SrNetoChan** [comment]
I got the same. I have install using `apt-get install ninja-build` on ubuntu 18.04.

The error occurs with or without arguments

**Comment by AmjadHD** [comment]
on windows too.

**Comment by mydongistiny** [comment]
Depending on what project you're building and what arguments you are using you may need to set

GYP_GENERATORS=ninja

**Comment by ThanushreeEG** [comment]
even i have come across this issue.
 when i was trying to download fuse 

and i get the same error when i give the command ninja or apt-get install ninja
![screenshot from 2018-10-10 21-22-06](https://user-images.githubusercontent.com/31587521/46749157-ca0f5100-ccd2-11e8-81c7-cd19f2ac6eb9.png)

**Comment by mydongistiny** [comment]
@ThanushreeEG what was the output of meson --reconfigure? Did it run ninja for you?

**Comment by jhasse** [comment]
@ThanushreeEG Does the `build.ninja` file exist in the CWD though? If not, this sound's like a Meson bug to me.

**Comment by jhasse** [comment]
To answer the original question:

> Why is this problem still occurring when you have successfully installed ninja?

Because Ninja needs a build definition file in the current working directory.

**Comment by hyperfraise** [comment]
Same problem on ubuntu 18.04

```damien@Earth2:~$ sudo ninja -v
ninja: error: loading 'build.ninja': No such file or directory
damien@Earth2:~$ meson --reconfigure
Error during basic setup:

Must specify at least one directory name.
```

**Comment by jhasse** [comment]
@hyperfraise It seems you're running meson in your home directory.

**Comment by Praveer1981** [comment]
I to encounter the same problem when I tried to build LLVM on windows.
> cmake -G ninja ../llvm

**Comment by hongyx11** [comment]
+1 tried on two mac and install from brew or source, all get error loading

**Comment by leonsaber** [comment]
I got how to work use `gn gen out/Default` , then you will find build.ninja file in /out/Default
run ninja under /out/Default

**Comment by imshengli** [comment]
> I got how to work use `gn gen out/Default` , then you will find build.ninja file in /out/Default
> run ninja under /out/Default

In this case, same error happened. `ninja: error: loading 'build.ninja': No such file or directory`
We can use `-f FILE` specify input build file.

**Comment by hanzhaogang** [comment]
> 
> 
> I got how to work use `gn gen out/Default` , then you will find build.ninja file in /out/Default
> run ninja under /out/Default

What is `gn`, and how can we get it?

**Comment by newmagnetar** [comment]
> What is `gn`, and how can we get it?

I got it from
https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up

**Comment by qingdujun** [comment]
same problem~

```
qingdujun@MacBook-Pro src % autoninja -C out/Release/ headless_shell
ninja: Entering directory `out/Release/'
ninja: error: loading 'build.ninja': No such file or directory
Traceback (most recent call last):
  File "/Users/qingdujun/repos/workspace/depot_tools/ninjalog_uploader_wrapper.py", line 118, in <module>
    sys.exit(main())
  File "/Users/qingdujun/repos/workspace/depot_tools/ninjalog_uploader_wrapper.py", line 76, in main
    config = LoadConfig()
  File "/Users/qingdujun/repos/workspace/depot_tools/ninjalog_uploader_wrapper.py", line 31, in LoadConfig
    'chromium-build-stats.appspot.com'),
  File "/Users/qingdujun/repos/workspace/depot_tools/ninjalog_uploader.py", line 44, in IsGoogler
    _, content = h.request('https://'+server+'/should-upload', 'GET')
  File "/Users/qingdujun/.vpython-root/867170/lib/python2.7/site-packages/httplib2/__init__.py", line 1659, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "/Users/qingdujun/.vpython-root/867170/lib/python2.7/site-packages/httplib2/__init__.py", line 1399, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/Users/qingdujun/.vpython-root/867170/lib/python2.7/site-packages/httplib2/__init__.py", line 1355, in _conn_request
    response = conn.getresponse()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1108, in getresponse
    raise ResponseNotReady()
httplib.ResponseNotReady
```

**Comment by qingdujun** [comment]
Building on Mac only works without headless.gn.

The reason for that is that the headless windowing system (ozone's headless platform in particular) that we use on Linux/Win and is configured in headless.gn isn't yet supported on Mac.


On Wed, Aug 9, 2017, 17:04 brandon.kobel via headless-dev <headle...@chromium.org> wrote:
When attempting to build the headless-shell on mac and performing the following step from Usage as a C++ library:

**Comment by Daniel34M** [comment]
The solution mentioned in #2290 worked for me!

**Comment by agalod** [comment]
Try running `gn gen out/Default`

**Comment by xiaozhiming2233** [comment]
the same problem on ubuntu 18.04

**Comment by twitchplaystrading** [comment]
Same on MX Linux 4.19.0-16-amd64 #1 SMP Debian 4.19.181-1. I've tried to use ninja on and debian and arch. Got it barely working on arch but no avail in debian.

**Comment by Tekh-ops** [comment]
Is there a fix for this

**Comment by erichodges** [comment]
OS X 11.3   I also have this issue. It's unclear to me how to fix this. I don't understand why this was closed as I don't see a clear fix above.

**Comment by Akhil564** [comment]
Hi, in the process of installing GTK stress testing tool  i need to do install dependencies. but when I am executing with ninja command i am getting an issue...anyone could you please help me out.. thanks in advance.

following are the dependencies need to install....
git clone --recurse-submodules -j4 https://gitlab.com/leinardi/gst.git
cd gst
git checkout release
pip3 install -r requirements.txt
meson . build --prefix /usr
ninja -v -C build
ninja -v -C build install
------------------------------------------------------------------------------------------------
getting an issue with ninja
#ninja -v -C build
ninja: Entering directory `build'
ninja: fatal: chdir to 'build' - No such file or directory

**Comment by ashok277** [comment]
> environment :Mac
> Install ninja with the brew intsall ninja command
> then on terminal execute ninja occur ninja: error: loading 'build.ninja': No such file or directory 。
> ➜ ~ ninja
> ninja: error: loading 'build.ninja': No such file or directory
> ➜ ~
> 
> Why is this problem still occurring when you have successfully installed ninja?

Did you try to locate the build.ninja file and run ninja in that folder?

**Comment by ThisIsMython** [comment]
this problem still there

**Comment by passos** [comment]
```
~ ❯ brew reinstall ninja
==> Downloading https://ghcr.io/v2/homebrew/core/ninja/manifests/1.10.2_1
Already downloaded: /Users/xxx/Library/Caches/Homebrew/downloads/3d885b06ee8943649e1e4f3edebcca2003dc2e9e332202cc2a0b16ccce4f8cae--ninja-1.10.2_1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/ninja/blobs/sha256:b3e0b14cffa2d227c0b4140c1a3742a2a4e7e3966a429e1d253b0e29acfb6293
Already downloaded: /Users/xxx/Library/Caches/Homebrew/downloads/ecdaa7d4e110c91481ede9ce38cf5a8fd3b025c3943ec3efefc1c4ff99b0fc6e--ninja--1.10.2_1.big_sur.bottle.tar.gz

~ ❯ ninja
ninja: error: loading 'build.ninja': No such file or directory
```

`macOS 11.6.1`

**Comment by gmolkho** [comment]
At U20 was resolved for me only when  using "pip install ninja" rather than "sudo apt install ninja-build"

**Comment by HriGrit** [comment]
> > I got how to work use `gn gen out/Default` , then you will find build.ninja file in /out/Default
> > run ninja under /out/Default
> 
> In this case, same error happened. `ninja: error: loading 'build.ninja': No such file or directory` We can use `-f FILE` specify input build file.

Can you show how the command line code will look like?

**Comment by razeek-j** [comment]
i got the solution 

just type :  sudo apt install generate-ninja

that's it

**Comment by LiWentomng** [comment]
> i got the solution
> 
> just type : sudo apt install generate-ninja
> 
> that's it

It works. Thanks.

**Comment by Gamer069** [comment]
> In case of build

i have the same problem, and i already did that ofc, the out/Default directory exists, gn gen out/Default didn't provide any output, autoninja -C out/Default complains about the build.ninja file not being a thing in the directory, (I double-checked, yeah it doesn't exist.), also i'm doing this on ubuntu.

**Comment by Gamer069** [comment]
> sudo apt install generate-ninja

for me that doesn't work.

**Comment by morebray** [comment]
Lösung für MACos
  cp  /Users/username/gst/build/build.ninja  /usr/local/Cellar/ninja/1.11.1/bin
dann
  ninja -v

**Comment by haciMMicah** [comment]
> > sudo apt install generate-ninja
> 
> that doesn't work for me.

You probably just have the generated `build.ninja` file in a different directory.

**Comment by IceAsteroid** [comment]
The output of meson setup build said the system lacks glib-compile-resources.

After installing glib-devel, I managed to run ninja -C build successfully.

**Comment by nguyenph194812** [comment]
is there a fix for this, i met this problem when running vcpkg installation
![image](https://github.com/ninja-build/ninja/assets/80508916/04cc77a0-ee49-45aa-af47-6ba051474294)

**Comment by tunm1228** [comment]
You try `gn gen out/debug --args='import("//with_pw_rpc.gni")'`
<img width="951" alt="image" src="https://github.com/ninja-build/ninja/assets/42457760/33ed8a5d-66fb-4cab-b5e0-cdaa9aaa224a">

**Comment by manojgit31792** [comment]
~/AndroidStudioProjects/COAT
$ sudo apt install generate-ninja
bash: sudo: command not found

I am getting this error when trying to install on android studio by using git bash. Can anyone help how to solve this?

**Comment by ramardoautistin** [comment]
i am going insane trying to compile aseprite on linux cause i keep getting this exact same error even after supposedly installing ninja :/

**Comment by LuftRev** [comment]
> i am going insane trying to compile aseprite on linux cause i keep getting this exact same error even after supposedly installing ninja :/

Likewise but on mac, I'm not a dev in any way shape or form so this all basically moon runes to me. Rather infuriating...

**Comment by visualblind** [comment]
> i got the solution
> 
> just type : sudo apt install generate-ninja
> 
> that's it

🥳 I confirm `sudo apt install generate-ninja` fixes the problem on Ubuntu 22.04/Mint 21.3 and generates the build.ninja file, but only after re-running `meson setup ...`. 

Maybe one of these days I should figure out what exactly meson and ninja actually are. I assume they're the equivalent of ./configure and make/make install.

**Comment by JacobRosenlund** [comment]
This might be a bit silly, but it probably won't hurt to put it out there. 

I had an issue with this exact error, and I fixed it by re-cloning the git repo I pulled from. After that, ninja played nicely.
